pkg/cephfs/volumemounter.go

```diff
@@ -116,10 +124,36 @@ func mountFuse(mountPoint string, cr *credentials, volOptions *volumeOptions, vo
 		return err
 	}
 
-	if !bytes.Contains(stderr, []byte("starting fuse")) {
+	// Parse the output:
```

Comment:
can you provide a sample output? It helps understand the parsing below. Thanks

Reply:
This doesn't output anything unfortunately, but i can sketch it out:

this is the stderr output from `ceph-fuse`
```
2019-02-26 21:30:52.556 7f0fb5589c00 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2019-02-26 21:30:52.568 7f0fb5589c00 -1 init, newargv = 0x1339e30 newargc=9
ceph-fuse[274]: starting ceph client
2019-02-26 21:30:52.568 7f0fb5589c00 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
ceph-fuse[274]: starting fuse
         ^   ^  ^
         |   |  |
----------------+ idx points to 's'
         |   |
-------------+ pidEnd points to ']'
         |
---------+ pidStart points to '['
```

First, we need to search for `starting fuse` string to make sure the mount is ok. The PID of the daemon is delimited by `[` and `]`: since we already have the starting position of that string in `idx`, and we know what the output _should_ look like, we can start searching for the right bracket, starting at the end of the string (as opposed to starting the search at the beginning of the output - this would be less "optimal"). Once we have the position of the right bracket in `pidEnd`, the same thing is repeated for the left bracket `[`, whose position is stored in `pidStart`. The PID should be in between `pidStart` and `pidEnd`, i.e. `stderr[pidStart+1 : pidEnd]`

Reply:
would a regex work? https://play.golang.org/p/U0Zo-qXlzfA

Reply:
Uff... this is a nice case of over-complicating things, isn't it? :P I'll send a fix